README.md

```diff
@@ -39,10 +50,17 @@ Version 1.2.0 of the plugin introduced improved logging for debugging purposes.
 
 You can also try chatting with us in the #gitlab-plugin channel on the Freenode IRC network.
 
-# Supported GitLab versions
+# Plugin Modes and Supported GitLab versions
+
+The Plugin supports two operation modes: **Modern** and **Legacy**. The main difference between these modes is how they deal with Merge Requests from GitLab. While the Legacy mode requires setting up multiple Git repositories and using parameterized builds, the Modern mode works with a single Git repository and relies on GitLab's availability of Merge Requests from the `origin` repository. The rest of this document is split for these two modes of operation (unless specified).
 
-* 7.14.x where it emulates Jenkins as a GitLabCI Web Service
-* 8.1.x and newer via the new commit status API that supports with external CI services like Jenkins
+The following versions of GitLab are supported by each of the modes:
+
+* Modern
+  * 8.1.x and newer
```

Comment:
If you are using **>= 8.1** **Modern** is definitely better IMO. That said, nothing prevents you to use the **Legacy** mode. While I'd argue against these, I think there are some scenarios where you might want to leave Legacy mode:
1. Don't want to (or can't) mess with your Jenkins setup.
2. You want Jenkins to build pushes _on forks_ (**Modern** will build MR and commits added to MR, not commits to forks)

Basically, Legacy still works and **Modern** is opt-in (it's not even the default, although I think it has big advantages). If we want to push people to use **Modern**, we can change the working here.